components/UI/Map/LeafletMap.tsx

```diff
@@ -189,13 +190,26 @@ function LeafletMap() {
         ]
       : DEFAULT_CENTER;
 
-  const defaultZoom = zoom
-    ? parseFloat(zoom as string)
-    : localCoordinatesURL
-    ? parseFloat(localCoordinatesURL.split("zoom=")[1].split("&")[0])
-    : device === "desktop"
-    ? DEFAULT_ZOOM
-    : DEFAULT_ZOOM_MOBILE;
+  useEffect(() => {
+    const zoomFromQuery = Number(zoom);
+    if (zoomFromQuery) {
+      setDefaultZoom(zoomFromQuery);
+      return;
+    }
+
+    const zoomParamFromURL = localCoordinatesURL
+      ?.split("zoom=")?.[1]
+      ?.split("&")?.[0];
+    if (zoomParamFromURL) {
+      setDefaultZoom(parseFloat(zoomParamFromURL));
+      return;
+    }
+
+    if (device == "mobile") {
+      setDefaultZoom(DEFAULT_ZOOM_MOBILE);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
not sure if i agree with disabling useEffect deps, especially when we are doing a refactor. What's the reason we need to do this?

Reply:
The motivation behind using an empty `useEffect` is that we need `ComponentDidMount` behavior here because if we add all the variables that we use in `useEffect` we'll need to add `zoom` search parameter and it'll trigger this useEffect hook according to the current `zoom` level.